src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpOnTypeFormattingPass.cs

```diff
@@ -171,5 +173,219 @@ private static int LineDelta(SourceText text, IEnumerable<TextChange> changes)
 
             return delta;
         }
+
+        private static List<TextChange> CleanupDocument(FormattingContext context, Range? range = null)
```

Comment:
Was this all a direct copy n paste or some changes ?

Reply:
Purely mechanical copy and paste in the first commit. I thought I'd use the "Pull Members Up" refactoring to do it, but turns out I needed "Push Members Down" which doesn't exist (I should probably complain to @ryzngard later...)
Change to private in the second commit.
Change the if in the third.